### PR TITLE
feat: add rso authentication

### PR DESCRIPTION
**Pull request by lucasMesquitaBorges**
## What this PR does?
- Adds browser link `Sign In` with RSO integration `/`, after logged in the `/accounts/me` will be called

## Further Features
- Improve README.md (setup and project changes)
- Use `X-Rate-Limit` headers

### CODE REVIEW

**Review by lesand — APPROVED** [review]

File: .env.example [hunk 1]
```diff
@@ -1,2 +1,4 @@
 APP_PORT=3000
+RG_SSO_CLIENT_ID=
```
Comment (xelaadryth) [inline .env.example]: Can call this `RSO_CLIENT_ID`, the terminology we're going for is calling these oauth clients "rso clients"

File: handlers/httpcontrollers/controller.go [hunk 2]
```diff
@@ -1,18 +1,40 @@
...
+	"os"
 	"riot-developer-proxy/internal/domain/entities"
 	"riot-developer-proxy/internal/domain/services"
+	"strings"
 
 	"github.com/labstack/echo/v4"
 )
 
 const INTERNAL_SERVER_ERROR_MESSAGE = "Internal Server Error"
+const SSO_PROVIDER_BASE_URI = "https://auth.riotgames.com"
```
Comment (xelaadryth) [inline handlers/httpcontrollers/controller.go]: Can call this `RSO_BASE_URI` instead

**Review by xelaadryth — APPROVED** [review]

File: handlers/httpcontrollers/controller.go (hunk 2 above)
Comment (lucasMesquitaBorges) [reply handlers/httpcontrollers/controller.go]: done!

File: .env.example (hunk 1 above)
Comment (lucasMesquitaBorges) [reply .env.example]: done!

**Pull request merged**